recipes/apr/all/conanfile.py

```diff
@@ -88,6 +95,10 @@ def _patch_sources(self):
             tools.replace_in_file(os.path.join(self._source_subfolder, "include", "apr.h.in"),
                                   "@osuuid@", "0")
 
+    def build_requirements(self):
+        if self.settings.os == "Macos":
+            self.build_requires("libtool/2.4.6")
```

Comment:
- Please move `build_requirements()` this before `source()`, otherwise it's hard to understand the flow of execution.

- Why the condition to require libtool is different than the one for calling buildconf (and I advice again to manually call managed autotools executables instead of relying on some blackbox)?

Reply:
That's still hanging in the air, to quite myself:

> Regarding calling autoreconf. Keep in mind it was recommended to use buildconf and not autoreconf. I'm fine with using either and am willing to change this to autoreconf. But wanted to throw this note about buildconf here before going ahead.

I do not mind calling autotools directly, but it goes against the apr maintainers suggestion. In the end with the official recipes here I do not know, go by upstream suggestion or is general style here to call autotools.